omod/src/main/java/org/openmrs/module/datafilter/filter/SubmissionFilter.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.datafilter.filter;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.OpenmrsObject;
+import org.openmrs.User;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.datafilter.impl.api.DataFilterService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class SubmissionFilter implements Filter {
+
+    private static final Logger log = LoggerFactory.getLogger(SubmissionFilter.class);
+
+
+    private DataFilterService dataFilterService;
+    private LocationService locationService;
+    private UserService userService;
+
+
+    public SubmissionFilter() {
+        this.dataFilterService = Context.getRegisteredComponents(DataFilterService.class).get(0);
+        this.locationService = Context.getLocationService();
+        this.userService = Context.getUserService();
+    }
+
+
+    /**
+     * @see Filter#init(FilterConfig)
+     */
+    @Override
+    public void init(FilterConfig filterConfig) {
+        if (log.isDebugEnabled()) {
+            log.debug("Initializing datafilter web filter....");
+        }
+    }
+
+    /**
+     * @see Filter#doFilter(ServletRequest, ServletResponse, FilterChain)
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        request.getRequestDispatcher("/admin/users/user.form").forward(request, response);
```

Comment:
I'm not sure if this approach is ideal, for instance I see a scenario where your code below that grants access fails the above request passed, did you test this scenario?

Why not just add a separate page for managing access? In general extension points aren't used for forms to capture data. If you want to use an extension point on a form, I would consider using ajax to send form data for your extension. It means this will only be done for an existing user.

Reply:
@wluyima Ideally I agree to the point that it would be better to have a separate page for managing user access. There was a business requirement that wanted it on the legacy ui for now and later expand this to be a different page altogether.
cc @mks-d   

We have tested the scenario where the user has not been created and in that case access is not granted to the location. We will test the scenario you mentioned and post back